tracee-ebpf/tracee/internal/bufferdecoder/decoder.go

```diff
@@ -0,0 +1,301 @@
+// Package bufferdecoder implements simple translation between byte
+// sequences and the user-defined structs.
+//
+// The package favors efficiency over flexibility. The provided API
+// allows fast decoding of byte sequence sent by the Tracee eBPF program from
+// kernel-space to user-space.
+package bufferdecoder
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/aquasecurity/tracee/pkg/external"
+)
+
+type EbpfDecoder struct {
+	buffer []byte
+	cursor uint32
```

Comment:
setting cursor as an int type would save the extra castings to uint32 done below, won't it?